### ISSUE TITLE
[bug] conan config install fails when URLs are not available anymore

### ISSUE THREAD

**Issue by GrimSqueaker** [conan-io/conan#7046]
### Environment Details
  * Operating System+version: Windows 10
  * Compiler+version: n/a
  * Conan version: 1.25.1
  * Python version: 3.6

### Steps to reproduce
```
conan config install some_url.zip
# remove the zip file
conan config install
```

### Context
The problem is even more severe. I tried to use the new auto install of configs and stumbled upon this problem. Because issuing almost any command triggers a config install when the timer has passed. This is related to https://github.com/conan-io/conan/pull/6824

Also:
- Can the order in which `conan config install` commands have been issued be preserved? Because in our case some configs might be overwritten by subsequent calls.
- How can entries in `config_install.json` be removed? Maybe even via a `conan config install`? For example we are planning to migrate from one server that hosts the configs to another. It would be good if the auto config install could also update the config URLs.

### Logs
```
PS D:\temp\conan> conan search
Traceback (most recent call last):
  File "conan\conans\client\command.py", line 2016, in run
  File "conan\conans\client\conan_api.py", line 92, in wrapper
  File "conan\conans\client\conan_api.py", line 608, in config_install
  File "conan\conans\client\conf\config_installer.py", line 234, in configuration_install
  File "conan\conans\client\conf\config_installer.py", line 201, in _process_config
  File "conan\conans\client\conf\config_installer.py", line 66, in _process_zip_file
  File "conan\conans\client\tools\files.py", line 104, in unzip
  File "zipfile.py", line 1204, in __init__
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\temp\\conan\\config-buildserver.zip'

ERROR: [Errno 2] No such file or directory: 'D:\\temp\\conan\\config-buildserver.zip'
```

**Comment by memsharded** [comment]
Thanks for reporting this issue. Every stack trace is indeed a bug, and this needs to be fixed.

- If an origin of ``config install`` is missing a clear error reporting this should be provided. I think nothing else should be done here, no automatic removal of it or the like, not continuing with a warning, but an error.
- At the moment, editing the ``config_install.json`` is the only way to remove them. Please do that as a workaround.
- It seems a ``conan config install-rm| --remove`` or something like that is necessary. I would suggest to have this possibility. Updating URLs sounds a bit more tricky, and maybe too much for Conan. For a planned migration I guess that a ``sed`` over the ``config_install.json`` could be good enough.
- Changing the order sounds also like a bug. It aims to preserve the order, and there are tests for that. I would love to have more details of something to reproduce this order bug.

**Comment by GrimSqueaker** [comment]
>     * If an origin of `config install` is missing a clear error reporting this should be provided. I think nothing else should be done here, no automatic removal of it or the like, not continuing with a warning, but an error.

True for `config install`. The follow up question is: How to handle this situation with `config_install_interval = xy` and other commands than `config install`? Perhaps, they should treat this as a warning and and skip the config update.

>     * At the moment, editing the `config_install.json` is the only way to remove them. Please do that as a workaround.
> 
>     * It seems a `conan config install-rm| --remove` or something like that is necessary. I would suggest to have this possibility. Updating URLs sounds a bit more tricky, and maybe too much for Conan. For a planned migration I guess that a `sed` over the `config_install.json` could be good enough.

Yes, for local workarounds this should do the trick. I thought I could simplify the future rollout of configurations and profiles to our 100 developers via this method. On the build servers I can use Saltstack.
You are right that maybe this is just out of scope for Conan since other configuration management tools exist. I just thought "Let's give this cool new feature in 1.25.0 a try" :-)
 
>     * Changing the order sounds also like a bug. It aims to preserve the order, and there are tests for that. I would love to have more details of something to reproduce this order bug.

To reproduce:
```
conan config install a.zip
conan config install b.zip
conan config install a.zip
conan config install
```
where a.zip and b.zip contain a `remotes.txt` with mutual exclusive remotes. Expected behaviour would be to have the remotes of a.zip after the final `config install`. But I get the remotes that are defined in b.zip, because the order in `conan_config.json` is not changed after the third line (the second `conan config install a.zip`).
Or am I wrong in assuming that the last issued URL should take precedence (sorted to the bottom in `conan_config.json`)?

**Comment by memsharded** [comment]
> True for config install. The follow up question is: How to handle this situation with config_install_interval = xy and other commands than config install? Perhaps, they should treat this as a warning and and skip the config update.

I still thing that for auto-install it is an error too. We should make sure that it is not re-firing the auto ``conan config install`` again and again, blocking a fix, but that is a broken setup that should be explicitly addressed. Same policy as when a remote is not accessible, it will error. You need to explicitly disable it, but it will not skip it and continue without user intervention.

> where a.zip and b.zip contain a remotes.txt with mutual exclusive remotes. Expected behaviour would be to have the remotes of a.zip after the final config install. But I get the remotes that are defined in b.zip, because the order in conan_config.json is not changed after the third line (the second conan config install a.zip).

I see. I think that the current behavior keeps the order of new origins, but if an existing origin is manually invoked again, individually, it doesn't re-define the existing order. In short: only new origins are appended to existing one, otherwise existing order is maintained. I am not sure if it is a bug, it is bad UX, or it desired behavior, to be honest (cc/ @jgsogo )

**Comment by jgsogo** [comment]
Hi! I see here different feature request and almost all of them can be implemented in different ways, all of them valid. I really don't have a strong opinion about them, but I think that both _config install origins_ and _remotes_ need the same set of commands and behavior. Now that we are thinking again about the CLI maybe we get some answer from that perspective given the proposed commands to handle them (@czoido )

Before the new command line maybe we need a way to list and remove entries in the `config.json` file, to avoid modifying that file by hand... but it would be different from existing commands for `remote`, something like `conan config install --list` and `conan config install item --remove`. 😕

**Comment by memsharded** [comment]
> conan config install --list and conan config install item --remove

Yes, I would go for that

**Comment by memsharded** [comment]
``conan config install --list and conan config install item --remove`` will be available in next 1.27 release

**Comment by GrimSqueaker** [comment]
Thanks.
